### ISSUE TITLE
NX doesn't throw the lint error when buildable library depends on itself. Lint passes without any errors. The build fails.

### ISSUE THREAD

**Issue by nikbabchenko** [nrwl/nx#16187]
### Current Behavior

Importing the buildable library from itself via '@some-lib-path' doesn't throw any lint errors.
Nx lint passes.
Build fails.

### Expected Behavior

Importing library from the same library should throw error - about relative paths. 
"**Projects should use relative imports to import from other files within the same project. Use "./path/to/file" instead of import**" (nx-enforce-boundaries)

### GitHub Repo

https://github.com/nrwl/nx-examples/pull/271

### Steps to Reproduce

1. Create buildable library for ex. 'ui-kit/core' (using nx-examples repo or the latest nx workspace - 15.9.2)
2. Import the module from the library inside the library -  for example 
<img width="572" alt="Screenshot 2023-04-08 at 14 53 14" src="https://user-images.githubusercontent.com/11876062/230722235-a069e485-f7b8-4878-b677-6bf02ccf6523.png">
3. Observe - there are no any lint errors.
<img width="540" alt="Screenshot 2023-04-08 at 14 55 32" src="https://user-images.githubusercontent.com/11876062/230722310-de29627d-c1de-49f1-94f9-6e10910e505b.png">

4. Try to build the library - observe 'circular dependency on itself error'.

<img width="664" alt="Screenshot 2023-04-08 at 15 06 05" src="https://user-images.githubusercontent.com/11876062/230722768-a9da5947-2ffc-4cbd-b1ab-6a8221a0af90.png">

### Nx Report

```shell
Node : 18.15.0
   OS   : darwin x64
   yarn : 1.22.19
   
   nx                      : 15.9.0-rc.2
   @nrwl/js                : 15.9.0-rc.2
   @nrwl/jest              : 15.9.0-rc.2
   @nrwl/linter            : 15.9.0-rc.2
   @nrwl/workspace         : 15.9.0-rc.2
   @nrwl/angular           : 15.9.0-rc.2
   @nrwl/cli               : 15.9.0-rc.2
   @nrwl/cypress           : 15.9.0-rc.2
   @nrwl/devkit            : 15.9.0-rc.2
   @nrwl/eslint-plugin-nx  : 15.9.0-rc.2
   @nrwl/react             : 15.9.0-rc.2
   @nrwl/tao               : 15.9.0-rc.2
   @nrwl/web               : 15.9.0-rc.2
   @nrwl/webpack           : 15.9.0-rc.2
   @nrwl/nx-cloud          : 15.3.3
   typescript              : 4.9.5
   ---------------------------------------
   Community plugins:
   @ngrx/component-store : 15.3.0
   @ngrx/effects         : 15.3.0
   @ngrx/entity          : 15.3.0
   @ngrx/router-store    : 15.3.0
   @ngrx/store           : 15.3.0
   @ngrx/store-devtools  : 15.3.0
```

### Failure Logs

```shell
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UiKitCoreCoreModule as UiKitNoCore } from '@ui-kit/core';

console.log(UiKitNoCore);

@NgModule({
  imports: [CommonModule],
})
export class UiKitCoreCoreModule {}
```

**Comment by meeroslav** [comment]
Thank you for reporting the error @nikbabchenko.

Please make sure to have a single lock file and that this lock file is in sync with package.json so that reproducing the issue is accurate.

Your `ui-kit-core-core` project contains `ng-package.json`, which points to `src/index.ts`. That makes this project have a duplicate entry point with the same TS path alias. If you remove the `ng-package.json`, you will see the expected error.

**Comment by nikbabchenko** [comment]
@meeroslav, need to mention, the ' ui-kit-core-core' has been generated via NX console. 
So, still seems like a bug.

**Comment by meeroslav** [comment]
You are correct. Sorry for being so fast with my conclusions.

**Comment by nikbabchenko** [comment]
Thanks for tracking this one, because having the NX in order to control boundaries in combination with 'Webstorm' or 'Idea' adds  issues with circular imports the library on itself.

NX doesn't handle this scenario. So, basically you have to additionally add lint rules to handle self imports.

**Comment by meeroslav** [comment]
I have created the fix. Unfortunately, you will have to be a bit more patient until it's released.

Thank you again for reporting it.

**Comment by wSedlacek** [comment]
@meeroslav I noticed a regression after #16367.

In my case I have a buildable angular library which has secondary entry points which import the primary entry point.

@flagship-create/async-store -> @flagship-create/async-store/indexeddb

When it tries to check the import of `@flagship-create/async-store` it is prevented from reading the `ng-package.json` at the project root and thus doesn't consider it a secondary entry point (which technically it is not) but it should still be valid for a secondary entrypoint to import the primary entrypoint so long as there isn't a cycle.

**Comment by meeroslav** [comment]
Hey @wSedlacek sorry for the late reply. Please create a separate issue and link it to this one.
